cms/djangoapps/contentstore/views/item.py

```diff
@@ -593,17 +593,27 @@ def _duplicate_item(parent_usage_key, duplicate_source_usage_key, user, display_
             runtime=source_item.runtime,
         )
 
+        handle_children = True
+        handle_parenting = True
+
+        if hasattr(dest_module, 'studio_post_duplicate'):
+            # Allow an XBlock to do anything fancy it may need to when duplicated from another block.
+            # These blocks may handle their own children or parenting if needed. Let them return booleans to
+            # let us know if we need to handle these or not.
+            handle_children, handle_parenting = dest_module.studio_post_duplicate(store, parent_usage_key, source_item)
+
         # Children are not automatically copied over (and not all xblocks have a 'children' attribute).
         # Because DAGs are not fully supported, we need to actually duplicate each child as well.
```

Comment:
@Kelketek @bradenmacdonald Is that still the case? Wouldn't it be cleaner to delegate the duplication to the modulestore?